sidecar/internal/service/reclaimspace.go

```diff
@@ -88,14 +110,34 @@ func (rs *ReclaimSpaceServer) NodeReclaimSpace(
 	pvName := req.GetPvName()
 	klog.Info(pvName)
 
-	// TODO: get following params using kubeClient and pvName
+	pv, err := rs.kubeClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+	var deviceSecret map[string]string
+	var volID string
+	if pv.Spec.CSI == nil {
+		volID = pv.Spec.CSI.VolumeHandle
+		if pv.Spec.CSI.NodeStageSecretRef != nil {
+			// Get the secrets from the k8s cluster
+			deviceSecret, err = kube.GetSecret(ctx, rs.kubeClient, pv.Spec.CSI.NodeStageSecretRef.Name, pv.Spec.CSI.NodeStageSecretRef.Namespace)
+			if err != nil {
+				klog.Errorf("Failed to get secret %v", err)
+				return nil, status.Error(codes.InvalidArgument, err.Error())
+			}
+		}
+	}
+
+	//TODO: Configure the stagingPath using cmdline args
+	var stagingPath string = "/var/lib/kubelet/plugins/kubernetes.io/csi/pv/"
 
 	csiReq := &csiReclaimSpace.NodeReclaimSpaceRequest{
-		VolumeId:          "",
+		VolumeId:          volID,
 		VolumePath:        "",
-		StagingTargetPath: "",
-		VolumeCapability:  &csi.VolumeCapability{},
-		Secrets:           map[string]string{},
+		StagingTargetPath: stagingPath + pvName,
+		VolumeCapability: &csi.VolumeCapability{
+			AccessMode: &csi.VolumeCapability_AccessMode{
+				Mode: k8sAccessModeToCSIAccessMode(pv.Spec.AccessModes[0]),
```

Comment:
you should not choose one access mode here. we need something similar to https://github.com/kubernetes-csi/external-provisioner/blob/61e5b2d16c6bc7d3dbb0da81579cd57503f6c971/vendor/github.com/kubernetes-csi/csi-lib-utils/accessmodes/access_modes.go#L29